### PR TITLE
Basic OpenID Connect support

### PR DESCRIPTION
**Pull request by ondrejvelisek**
I need to deploy it on Elixir instance till wednesday. I'm not sure how does it work but we can deploy it just for this week and than undeploy it. (Deploy from particular branch?)

We can discuss it further tmrw.

### CODE REVIEW

File: perun-base/src/main/java/cz/metacentrum/perun/core/api/PerunClient.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        return true;
+    }
+
+    public enum Scope {
+
+        /* OAuth 2.0 custom scopes */
+        // None yet
+
+        /* OIDC custom claims */
+        BONA_FIDE_STATUS,
```
Comment (licehammer) [inline perun-base/src/main/java/cz/metacentrum/perun/core/api/PerunClient.java]: Should be configurable.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java [hunk 2]
```diff
@@ -310,6 +312,24 @@ else if (extLogin == null && req.getAttribute("SSL_CLIENT_VERIFY") != null && ((
...
 
+	protected PerunClient setupPerunClient(HttpServletRequest req) throws InternalErrorException, RpcException {
+
+		if (req.getHeader("OIDC_CLAIM_audience") != null && !req.getHeader("OIDC_CLAIM_audience").isEmpty()) {
+
+			String clientId = req.getHeader("OIDC_CLAIM_audience");
+
+			List<PerunClient.Scope> scopes = new ArrayList<>();
+			for (String scope : req.getHeader("OIDC_CLAIM_scopes").split(",")) {
+				scopes.add(PerunClient.Scope.valueOf(scope.trim().toUpperCase()));
```
Comment (licehammer) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java]: Might throw IllegalArgumentException

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/ApiCaller.java [hunk 3]
```diff
@@ -229,6 +232,13 @@ public VOOT getVOOTManager(){
 		return vootManager;
 	}
 
+	public OIDC getOIDCManager(){
+		if(oidcManager == null){
+			oidcManager = new OIDC();
```
Comment (licehammer) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/ApiCaller.java]: Susceptible to race condition

**Comment by ondrejvelisek** [comment]
We deployed it temporally from ondrejvelisek/oidc branch and merge will wait. Meanwhile I'm closing it.

**Pull request closed without merging**